src/typing/type.ml

```diff
@@ -1981,7 +1982,7 @@ and Object : sig
   (* A union type resolves to a resolved spread with more than one element *)
   and resolved = slice Nel.t
 
-  and slice = reason * props * dict * TypeTerm.flags
+  and slice = reason * props * dict * TypeTerm.flags * Properties.t
```

Comment:
I added original props (raw_props) because I needed polarity from field structure, and also doesn't wanted to get `mixed` on write-only properties